### PR TITLE
init commit.. function pointer for IntType working

### PR DESCRIPTION
**Pull request by CultureBunkerRecordings**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -225,6 +235,20 @@ IntType& IntType::operator*=(const int other)
...
+    intFunc(*intTypeHeap);
+    return *this;
+}
+
+IntType& IntType::apply(void(*intFunc) (int&))
+{
+    if(intFunc)
+    intFunc(*intTypeHeap);
+    return *this;
+}
```
Comment (matkatmusic) [inline main.cpp]: - [ ] Fix the Formatting
Also, what does your `std::function` return?

File: main.cpp [hunk 2]
```diff
@@ -291,6 +315,20 @@ FloatType& FloatType::operator*=(const float other)
...
+    if(floatFunc)
+    floatFunc(*floatTypeHeap);
+    return *this;
+}
+
+FloatType& FloatType::apply(void(*floatFunc) (float&))
+{
+    if(floatFunc)
+    floatFunc(*floatTypeHeap);
+    return *this;
```
Comment (matkatmusic) [inline main.cpp]: - [ ] Fix the Formatting
Also, what does your `std::function` return?

File: main.cpp [hunk 3]
```diff
@@ -357,6 +395,20 @@ DoubleType& DoubleType::operator*=(const double other)
...
+    if(doubleFunc)
+    doubleFunc(*doubleTypeHeap);
+    return *this;
+}
+
+DoubleType& DoubleType::apply(void(*doubleFunc) (double&))
+{
+    if(doubleFunc)
+    doubleFunc(*doubleTypeHeap);
+    return *this;
```
Comment (matkatmusic) [inline main.cpp]: - [ ] Fix the Formatting
Also, what does your `std::function` return?

**Review by matkatmusic — CHANGES_REQUESTED** [review]
Good, but you have to remember to make your code look good for other people who will look at it.

File: main.cpp (hunk 1 above)
Comment (CultureBunkerRecordings) [reply main.cpp]: it returns a reference to intType, to allow chaining

File: main.cpp [hunk 4]
```diff
@@ -491,23 +503,26 @@ int main()
 
     std::cout << "Plus 10 to i using a function pointer = " << i << std::endl;
 
-    i.apply([](int& intHeap){ intHeap += 10; });
+    i.apply([&](int& intHeap) -> IntType& { intHeap += 10;
+    return i; });
```
Comment (matkatmusic) [inline main.cpp]: - [ ] Don't capture more than you need to in your lambda.

File: main.cpp [hunk 5]
```diff
@@ -491,23 +503,26 @@ int main()
...
 
     std::cout << "Plus 10 to i using a lambda = " << i << std::endl;
 
     f.apply(plusTen);
 
     std::cout << "Plus 10 to f using a function pointer = " << f << std::endl;
 
-    f.apply([](float& floatHeap){ floatHeap += 10; });
+    f.apply([&](float& floatHeap) -> FloatType& { floatHeap += 10;
```
Comment (matkatmusic) [inline main.cpp]: - [ ] Don't capture more than you need to in your lambda.

File: main.cpp [hunk 6]
```diff
@@ -491,23 +503,26 @@ int main()
...
+    return f; });
 
     std::cout << "Plus 10 to f using a lambda = " << f << std::endl;
 
     d.apply(plusTen);
 
     std::cout << "Plus 10 to d using a function pointer = " << d << std::endl;
 
-    d.apply([](double& doubleHeap){ doubleHeap += 10.0; });
+    d.apply([&](double& doubleHeap) -> DoubleType& { doubleHeap += 10.0;
```
Comment (matkatmusic) [inline main.cpp]: - [ ] Don't capture more than you need to in your lambda.

File: main.cpp [hunk 7]
```diff
@@ -238,14 +238,18 @@ IntType& IntType::operator*=(const int other)
 IntType& IntType::apply(std::function<IntType&(int&)> intFunc)
 {
     if(intFunc)
-    intFunc(*intTypeHeap);
+    {
+        intFunc(*intTypeHeap);
```
Comment (matkatmusic) [inline main.cpp]: `intFunc` returns `IntType&`.   That is the same thing this `apply` function returns. 
- [ ] return the result of calling the std::function if the std::function is valid.

File: main.cpp [hunk 8]
```diff
@@ -318,15 +322,19 @@ FloatType& FloatType::operator*=(const float other)
 FloatType& FloatType::apply(std::function<FloatType&(float&)> floatFunc)
 {
     if(floatFunc)
-    floatFunc(*floatTypeHeap);
+    {
+        floatFunc(*floatTypeHeap);
```
Comment (matkatmusic) [inline main.cpp]: - [ ] return the result of calling the std::function if the std::function is valid.

File: main.cpp [hunk 9]
```diff
@@ -398,15 +406,19 @@ DoubleType& DoubleType::operator*=(const double other)
 DoubleType& DoubleType::apply(std::function<DoubleType&(double&)> doubleFunc)
 {
     if(doubleFunc)
-    doubleFunc(*doubleTypeHeap);
+    {
+        doubleFunc(*doubleTypeHeap);
```
Comment (matkatmusic) [inline main.cpp]: - [ ] return the result of calling the std::function if the std::function is valid.

**Review by matkatmusic — CHANGES_REQUESTED** [review]

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**